### PR TITLE
Assorted clean up

### PR DESCRIPTION
**Pull request by erik-stripe**
This commit makes a few changes:

 1. Removes unused file.
 2. Fixes version typo in the README.
 3. Add a bit to the README
 4. Improves FunctionK memoization

### CODE REVIEW

File: core/src/main/scala/com/stripe/dagon/Memoize.scala [hunk 1]
```diff
@@ -39,14 +39,22 @@ object Memoize {
...
-   * Memoize a FunctionK using an HCache internally.
+   * Memoize a FunctionK using an HMap internally.
    */
   def functionK[A[_], B[_]](f: RecursiveK[A, B]): FunctionK[A, B] = {
-    val hcache = HCache.empty[A, B]
-    lazy val hg: FunctionK[A, B] = new FunctionK[A, B] {
-      def toFunction[T] = { at =>
-        hcache.getOrElseUpdate(at, f((at, hg)))
+    var hmap = HMap.empty[A, B]
+    def getOrElseUpdate[T](at: A[T], bt: => B[T]): B[T] =
```
Comment (johnynek) [inline core/src/main/scala/com/stripe/dagon/Memoize.scala]: this is just a re-implementation of HCache, no?

File: core/src/main/scala/com/stripe/dagon/HCache.scala [hunk 2]
```diff
@@ -6,16 +6,6 @@ package com.stripe.dagon
 sealed class HCache[K[_], V[_]] private (init: HMap[K, V]) {
   private var hmap: HMap[K, V] = init
 
-  /**
```
Comment (johnynek) [inline core/src/main/scala/com/stripe/dagon/HCache.scala]: any real reason to remove these? They seem valuable to me and very easy to add to the API?

File: core/src/main/scala/com/stripe/dagon/DependantGraph.scala [hunk 3]
```diff
@@ -1,57 +0,0 @@
...
-  def dependantsOf(p: T): Option[List[T]] =
-    if (isNode(p)) Some(graph(p).toList) else None
-
-  def fanOut(p: T): Option[Int] = dependantsOf(p).map { _.size }
-
-  /**
-   * Return all dependendants of a given node.
-   * Does not include itself
-   */
-  def transitiveDependantsOf(p: T): List[T] = depthFirstOf(p)(graph)
```
Comment (johnynek) [inline core/src/main/scala/com/stripe/dagon/DependantGraph.scala]: can we add transitiveDependents to ExpressionDag if we remove this.

File: core/src/main/scala/com/stripe/dagon/DependantGraph.scala [hunk 4]
```diff
@@ -1,57 +0,0 @@
...
-   */
-  private lazy val graph: NeighborFn[T] = reversed(nodes)(dependenciesOf(_))
-
-  private lazy val depths: Map[T, Int] = dagDepth(nodes)(dependenciesOf(_))
-
-  /**
-   * The max of zero and 1 + depth of all parents if the node is the graph
-   */
-  def isNode(p: T): Boolean = nodeSet.contains(p)
-  def depth(p: T): Option[Int] = depths.get(p)
```
Comment (johnynek) [inline core/src/main/scala/com/stripe/dagon/DependantGraph.scala]: can we add this to ExpressionDag if we remove this here? We aren't using it, but summingbird and scalding to use this code, so removing it I think is a bummer.

**Comment by erik-stripe** [comment]
We spoke and have a plan to rework this PR, porting functionality from `DependentGraph` to `ExpressionDag`, renaming to that to e.g. `Dag`, and adding `Cache` to parallel `HCache` and reworking `Memoize` to use both (with parallel implementations).

File: core/src/main/scala/com/stripe/dagon/HCache.scala (hunk 2 above)
Comment (non) [inline core/src/main/scala/com/stripe/dagon/HCache.scala]: we weren't using or testing them, so i removed them. i can readd them with tests if we expect folks to use these types themselves (i am not sure i expect other people to do that beyond e.g. `Memoize`).

**Pull request closed without merging**